packages/grid/src/vaadin-grid-data-provider-mixin.js

```diff
@@ -273,7 +273,7 @@ export const DataProviderMixin = (superClass) =>
       el.index = index;
       const { cache, scaledIndex } = this._cache.getCacheAndIndex(index);
       const item = cache.items[scaledIndex];
-      if (item) {
+      if (item !== undefined) {
```

Comment:
Without this change, the grid falls into an infinite loop if one of the items is `0`.